activerecord/lib/active_record/railtie.rb

```diff
@@ -32,6 +32,7 @@ class Railtie < Rails::Railtie # :nodoc:
 
     config.active_record.use_schema_cache_dump = true
     config.active_record.maintain_test_schema = true
+    config.active_record.require_belongs_to = true
```

Comment:
Since this is in the Railtie, it'll surprisingly affect apps that upgrade to newer Rails. We'll need to generate new apps with it set, but use `false` as the default to preserve existing app behavior.